### PR TITLE
node: always start blocksync and avoid misconfiguration

### PR DESCRIPTION
**Pull request by tychoish**
Addresses: #8899

### CODE REVIEW

**Comment by williambanfield** [comment]
Is blocksync intended to always be enabled in v0.35? It looks like the option is still present to disable it.

**Comment by tychoish** [comment]
> Is blocksync intended to always be enabled in v0.35? It looks like the option is still present to disable it.

I can't see a reason why you'd ever want to disable it.

File: node/node.go [hunk 1]
```diff
@@ -700,10 +700,8 @@ func (n *nodeImpl) OnStart() error {
 	}
 
 	if n.config.Mode != config.ModeSeed {
-		if n.config.BlockSync.Version == config.BlockSyncV0 {
```
Comment (creachadair) [inline node/node.go:703]: Maybe at this point we should just remove this setting from the config file entirely? At least at master and maybe in v0.36 that seems safe.

Comment (tychoish) [reply node/node.go:703]: I believe it's out in master and 0.36 this is just against 0.35

**Review by creachadair — APPROVED** [review]

**Comment by williambanfield** [comment]
I don't think we should change this in the released version.

**Pull request merged**